From b834507636273aec6ac6735a8dfce566e8083961 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 23 Jun 2020 17:17:13 +0800
Subject: [PATCH 04/30] HACK: backend-drm: Disable atomic and
 DRM_CLIENT_CAP_ASPECT_RATIO

These APIs are unstable on Rockchip BSP 4.4 kernel.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/backend-drm/kms.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libweston/backend-drm/kms.c b/libweston/backend-drm/kms.c
index 192435c..8101921 100644
--- a/libweston/backend-drm/kms.c
+++ b/libweston/backend-drm/kms.c
@@ -1489,6 +1489,8 @@ init_kms_caps(struct drm_backend *b)
 	weston_log("DRM: %s universal planes\n",
 		   b->universal_planes ? "supports" : "does not support");
 
+	/* HACK: Atomic API has issues on Rockchip BSP 4.4 kernel */
+#if 0
 	if (b->universal_planes && !getenv("WESTON_DISABLE_ATOMIC")) {
 		ret = drmGetCap(b->drm.fd, DRM_CAP_CRTC_IN_VBLANK_EVENT, &cap);
 		if (ret != 0)
@@ -1498,6 +1500,7 @@ init_kms_caps(struct drm_backend *b)
 	}
 	weston_log("DRM: %s atomic modesetting\n",
 		   b->atomic_modeset ? "supports" : "does not support");
+#endif
 
 	ret = drmGetCap(b->drm.fd, DRM_CAP_ADDFB2_MODIFIERS, &cap);
 	if (ret == 0)
@@ -1516,10 +1519,13 @@ init_kms_caps(struct drm_backend *b)
 	if (!b->atomic_modeset || getenv("WESTON_FORCE_RENDERER"))
 		b->sprites_are_broken = true;
 
+	/* HACK: This conflicts with custom cap in Rockchip BSP 4.4 kernel */
+#if 0
 	ret = drmSetClientCap(b->drm.fd, DRM_CLIENT_CAP_ASPECT_RATIO, 1);
 	b->aspect_ratio_supported = (ret == 0);
 	weston_log("DRM: %s picture aspect ratio\n",
 		   b->aspect_ratio_supported ? "supports" : "does not support");
+#endif
 
 	return 0;
 }
-- 
2.20.1

